gpytorch/lazy/identity_lazy_tensor.py

```diff
@@ -0,0 +1,188 @@
+        self._device = device
+
+    @property
+    def batch_shape(self):
+        """
+        Returns the shape over which the tensor is batched.
+        """
+        return self._batch_shape
+
+    @property
+    def dtype(self):
+        return self._dtype
+
+    @property
+    def device(self):
+        return self._device
+
+    def _maybe_reshape_rhs(self, rhs):
+        if self._batch_shape != rhs.shape[:-2]:
+            batch_shape = _mul_broadcast_shape(rhs.shape[:-2], self._batch_shape)
+            return rhs.expand(*batch_shape, *rhs.shape[-2:])
+        else:
+            return rhs
+
+    @cached(name="cholesky", ignore_args=True)
+    def _cholesky(self, upper=False):
+        return self
+
+    def _cholesky_solve(self, rhs):
+        return self._maybe_reshape_rhs(rhs)
+
+    def _expand_batch(self, batch_shape):
+        return IdentityLazyTensor(
+            diag_shape=self.diag_shape, batch_shape=batch_shape, dtype=self.dtype, device=self.device
+        )
+
+    def _getitem(self, row_index, col_index, *batch_indices):
+        # Special case: if both row and col are not indexed, then we are done
+        if _is_noop_index(row_index) and _is_noop_index(col_index):
+            if len(batch_indices):
+                new_batch_shape = _compute_getitem_size(self, (*batch_indices, row_index, col_index))[:-2]
+                res = IdentityLazyTensor(
+                    diag_shape=self.diag_shape, batch_shape=new_batch_shape, dtype=self._dtype, device=self._device
+                )
+                return res
+            else:
+                return self
+
+        return super()._getitem(row_index, col_index, *batch_indices)
+
+    def _matmul(self, rhs):
+        return self._maybe_reshape_rhs(rhs)
+
+    def _mul_constant(self, constant):
+        return ConstantDiagLazyTensor(self.diag_values * constant, diag_shape=self.diag_shape)
+
+    def _mul_matrix(self, other):
+        return other
+
+    def _permute_batch(self, *dims):
+        batch_shape = self.diag_values.permute(*dims, -1).shape[:-1]
+        return IdentityLazyTensor(
+            diag_shape=self.diag_shape, batch_shape=batch_shape, dtype=self._dtype, device=self._device
+        )
+
+    def _prod_batch(self, dim):
+        batch_shape = list(self.batch_shape)
+        del batch_shape[dim]
+        return IdentityLazyTensor(
+            diag_shape=self.diag_shape, batch_shape=torch.Size(batch_shape), dtype=self.dtype, device=self.device
+        )
+
+    def _root_decomposition(self):
+        return self.sqrt()
+
+    def _root_inv_decomposition(self, initial_vectors=None):
+        return self.inverse().sqrt()
+
+    def _size(self):
+        return torch.Size([*self._batch_shape, self.diag_shape, self.diag_shape])
+
+    def _t_matmul(self, rhs):
+        return self._maybe_reshape_rhs(rhs)
+
+    def _transpose_nonbatch(self):
+        return self
+
+    def abs(self):
+        return self
+
+    def inverse(self):
+        return self
+
```

Comment:
exp could return self and log could return a ZeroLazyTensor I guess?

Reply:
Done.